src/fheroes2/editor/editor_object_popup_window.cpp

```diff
@@ -64,7 +66,35 @@ namespace
 
             // This is an invalid object!
             assert( 0 );
-            return "Invalid resource";
+            break;
+        }
+        case MP2::OBJ_ARTIFACT: {
+            for ( const auto & info : Maps::getObjectsByGroup( Maps::ObjectGroup::ADVENTURE_ARTIFACTS ) ) {
+                assert( !info.groundLevelParts.empty() );
+
+                if ( info.objectType == MP2::OBJ_ARTIFACT && info.groundLevelParts.front().icnIndex == tile.getMainObjectPart().icnIndex
+                     && info.groundLevelParts.front().icnType == tile.getMainObjectPart().icnType ) {
+                    return MP2::StringObject( MP2::OBJ_ARTIFACT ) + std::string( " - " ) + Artifact( static_cast<int32_t>( info.metadata[0] ) ).GetName();
```

Comment:
How about printing the name of the Artifact on the next line?
![изображение](https://github.com/user-attachments/assets/db9b4001-ec7d-4221-83de-e52ab6cbedb9)
For the long-named artifacts it looks not good when the part of it's name is moved to the next line:
![изображение](https://github.com/user-attachments/assets/7e80aa29-760b-4aab-992b-151738693d70)

```suggestion
                    return MP2::StringObject( MP2::OBJ_ARTIFACT ) + std::string( "\n" ) + Artifact( static_cast<int32_t>( info.metadata[0] ) ).GetName();
```

Reply:
Hi @Districh-ru , I followed random artifacts and monsters naming. For example, Random Monster - weak. I also need to check whether the window has enough space for this idea.